oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -340,6 +343,23 @@ public AccessToken refreshAccessToken() throws IOException {
     HttpRequest request = requestFactory.buildPostRequest(new GenericUrl(tokenServerUri), content);
     request.setParser(new JsonObjectParser(jsonFactory));
 
+    request.setIOExceptionHandler(new HttpBackOffIOExceptionHandler(new ExponentialBackOff()));
+    request.setUnsuccessfulResponseHandler(
+        new HttpBackOffUnsuccessfulResponseHandler(new ExponentialBackOff()).setBackOffRequired(
+            new BackOffRequired() {
+              public boolean isRequired(HttpResponse response) {
+                int code = response.getStatusCode();
+                return (
+                    // Server error --- includes timeout errors, which use 500 instead of 408
+                    code / 100 == 5
```

Comment:
Here I was just trying to be consistent with https://developers.google.com/api-client-library/java/google-http-java-client/reference/1.20.0/com/google/api/client/http/HttpBackOffUnsuccessfulResponseHandler.BackOffRequired#ON_SERVER_ERROR .

I would argue that, given the specification, the robust alternative would need to be `code >= 500 && code < 600`, which seems less readable as well as slower.